### ISSUE TITLE
Signup flow: sort groups by distance to the user

### ISSUE THREAD

**Issue by riyasinha** [yunity/karrot-frontend#912]
This feature would help the user to filter the different groups based on his/her preferences. 
We could have the following parameters to begin with:
- Location detector - geographical location, place from which food was recently picked ( this could be done with help of a map feature pulled [#903](https://github.com/yunity/karrot-frontend/pull/903) )
- Groups in which activities have recently taken place
- Category of the food - perishables, non-perishables 

@snehalgupta

**Comment by djahnie** [comment]
This issue seems to be similar to #877. If I understand it right, this proposal is an extension of before-mentioned issue and envisages more detailed input options than only a search bar.
- Location is adressed by #877 already
- Recent activity would need to be specified in the first place. How do you imagine this to work in detail? What counts as activity and which time is recent?
- Interesting split! But I suspect that there won't be many cases where _only_ non-perishables would be to save. Still, that doesn't matter that this doesn't make sense.

**Comment by riyasinha** [comment]
We opened this issue on the suggestion of @tiltec. He wanted us to describe our proposal and asked to work with @D0nPiano and implement his map feature. 
So this is what we had in mind - 
- Location Filter -  we can work with issue #877. Sorry for the ignorance.
- Recent Activities - We could create a system of points where when a group updates a valid activity, they earn some points. By this we can get a timestamp of the activity and add the feature to the filtering menu. This system of points could be further discussed into another direction altogether of rewards.
- Perishables/ Non-Perishables - I was thinking along the lines of the country I belong to, India. In the Indian sub-continent and in most of South-East Asia, there is a huge concept of Vegetables and Fish markets, where only such perishables are sold. I have had conversations with vendors and they have complained that many a times the items aren't sold and are wasted. Therefore, we wanted to focus on those groups and help the interested user to join such groups.

@snehalgupta

**Comment by D0nPiano** [comment]
I am unsure how to tackle recent activities and perishables - especially since the required information is not ready to be gathered/assessed anywhere as far as I know. Especially activity would also be interesting for other tasks though (e.g. #868 or #355) - Probably @tiltec or @nicksellen have more insight on what the current status of user activity is backend wise. The point system also sounds like it goes in the direction of self-regulated hierarchies #878

Map wise, once #903 is ready, it would be good to enable filtering when the map location / zoom level is changed. Once that is done, it should be relatively easy to enable filtering groups by location (entering a term in the search bar could just open the location on the map -> filteres groups automatically). Besides the other possible next tasks that i wrote down in #903, it would be great if you guys work on the filter after the PR is done.

**Comment by tiltec** [comment]
These are quite some topics for one issue. My proposal is to get started with a feature that sorts the groups by distance to the user:

> The groups could be sorted by distance from the user. You could get a guess of the users' location by GeoIP lookup or by using the location API of browsers. You can implement it in our Android app too.

The other topics don't seem to be connected to this, so my proposal is to first work on the location filter and handle the rest after that.

**Comment by nicksellen** [comment]
(related to recent activities point: https://github.com/yunity/karrot-frontend/issues/915 addresses hiding groups that are inactive)

**Comment by tiltec** [comment]
This topic seems interesting, but I assume it's rather low priority. I'll reopen if need arises  (e.g. too many groups on a single instance).
